### PR TITLE
feat(flags): Add BSD file flags

### PR DESCRIPTION
**Pull request by cfxegbert**
Add BSD file flags.  See the man page for chflags(1).

### CODE REVIEW

File: src/options/help.rs [hunk 1]
```diff
@@ -59,8 +59,9 @@ LONG VIEW OPTIONS
   -H, --links                list each file's number of hard links
   -i, --inode                list each file's inode number
   -m, --modified             use the modified timestamp field
-  -M, --mounts               show mount details (Linux and MacOS only)
+  -M, --mounts               show mount details (Linux and Mac only)
   -n, --numeric              list numeric user and group IDs
+  -O, --flags                list file flags (Mac and BSD only)
```
Comment (cfxegbert) [inline src/options/help.rs:64]: Used `-O` because the is was Mac ls uses.

**Review by MartinFillon — CHANGES_REQUESTED** [review]
This needs some screens to show behaviour

**Comment by cfxegbert** [comment]
> This needs some screens to show behaviour

```
target/debug/eza -lOh .                                                                                                       ─╯
Permissions Size User   Flags         Date Modified Name
drwxr-xr-x@    - rminsk -             27 Sep 15:56  benches
.rw-r--r--@ 3.9k rminsk -             27 Sep 15:56  build.rs
.rw-r--r--@  43k rminsk -             30 Nov 23:44  Cargo.lock
.rw-r--r--@ 3.2k rminsk -             30 Nov 23:44  Cargo.toml
.rw-r--r--@  30k rminsk -             30 Nov 23:44  CHANGELOG.md
.rw-r--r--@ 2.5k rminsk -              1 Nov 22:21  cliff.toml
.rw-r--r--@ 5.5k rminsk -             28 Aug 17:46  CODE_OF_CONDUCT.md
drwxr-xr-x@    - rminsk -              1 Nov 22:21  completions
.rw-r--r--@ 8.1k rminsk -             18 Nov 21:20  CONTRIBUTING.md
.rw-r--r--@ 1.7k rminsk -             28 Aug 17:46  deb.asc
.rw-r--r--@  12k rminsk -             19 Sep 22:08  deny.toml
drwxr-xr-x@    - rminsk -             18 Nov 21:20  devtools
drwxr-xr-x@    - rminsk -             18 Nov 21:20  docs
.rw-r--r--@ 7.8k rminsk -             18 Nov 21:20  flake.lock
.rw-r--r--@ 9.0k rminsk -             18 Nov 21:20  flake.nix
drwxr-xr-x@    - rminsk nodump        19 Nov 23:16  foo
drwxr-xr-x@    - rminsk nodump,hidden  7 Sep 19:44  hidden
```

**Review by MartinFillon — CHANGES_REQUESTED** [review]
The implementation is good on my side but waiting for @cafkafk answer regarding bsd action on element

**Comment by cfxegbert** [comment]
Adding support for Windows attributes.  I will need someone else to test it.

**Comment by MartinFillon** [comment]
I can test this tell me when this is ready to test

**Comment by cafkafk** [comment]
> The implementation is good on my side but waiting for @cafkafk answer regarding bsd action on element

not sure what you're referring to here, was this on matrix?

**Comment by MartinFillon** [comment]
yes

**Review by MartinFillon — DISMISSED** [review]
This works on windows 🟢

**Review by cafkafk — APPROVED** [review]
Code is spotless, and very neat. 

Only failing test was one for help which I regenerated.

This feature is very very cool, turns out non-linux users can have nice things too!

**Pull request merged**